### PR TITLE
Third-party tools with subdomain hosting

### LINKED ISSUE #27140: Hosting in subdirectory not broadly supported.
This is particular to 'Calling gRPC services hosted in a sub-directory' solution. 

There was a big surprise for me, and I have a feeling for other implementors as well, that I would hope is mentioned somewhere. While 'SubdirectoryHandler' solves the issue for .net clients, it appears that hosting in subdirectory is generally not supported by other clients: Postman, grpcurl (https://github.com/fullstorydev/grpcurl/issues/318). Id don't know if/how other clients/libraries approach it.  Effectively for universal compatibility, gRPC service has to be deployed under dedicated subdomains. I just wish it was mentioned somewhere.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: c730b5f9-031e-beab-b838-6e2f8befde30
* Version Independent ID: 84ddcb67-03be-655c-470a-f43a5152606f
* Content: [Troubleshoot gRPC on .NET Core](https://learn.microsoft.com/en-us/aspnet/core/grpc/troubleshoot?view=aspnetcore-6.0#feedback)
* Content Source: [aspnetcore/grpc/troubleshoot.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/grpc/troubleshoot.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-grpc**
* GitHub Login: @JamesNK
* Microsoft Alias: **jamesnk**

### PR DESCRIPTION
**Pull request by lucasmaj**
[EDIT by guardrex to add the "fixes" ... note @lucasmaj that this will auto-close the issue when the PR merges.]

Fixes #27140

[Internal doc build review](https://review.learn.microsoft.com/en-us/aspnet/core/grpc/troubleshoot?view=aspnetcore-7.0&branch=pr-en-us-27174)

### CODE REVIEW

**Comment by guardrex** [comment]
*Whooops!* I just noticed on the issue, @wadepickett, that you assigned yourself prior to Rick asking @lucasmaj to send this in. I didn't notice it until just now.

File: aspnetcore/grpc/troubleshoot.md [hunk 1]
```diff
@@ -170,6 +170,9 @@ The WPF application can use the gRPC generated types from the new class library
 
 ## Calling gRPC services hosted in a sub-directory
 
+> [!WARNING]
+> Many third-party gRPC tools don't support services hosted in subdomains. Consider finding a way to host gRPC as the root directory.
```
Comment (JamesNK) [inline aspnetcore/grpc/troubleshoot.md:174]:
```suggestion
> Many third-party gRPC tools don't support services hosted in subdirectories. Consider finding a way to host gRPC as the root directory.
```
> Many third-party gRPC tools don't support services hosted in subdomains.

subdomains are supported. It's services under a subdirectory that aren't.

image

File: aspnetcore/grpc/troubleshoot.md [hunk 2]
```diff
@@ -435,6 +438,9 @@ The WPF application can use the gRPC generated types from the new class library
...
```
Comment (JamesNK) [inline aspnetcore/grpc/troubleshoot.md:442]:
```suggestion
> Many third-party gRPC tools don't support services hosted in subdirectories. Consider finding a way to host gRPC as the root directory.
```

File: aspnetcore/grpc/troubleshoot.md [hunk 3]
```diff
@@ -650,6 +656,9 @@ The WPF application can use the gRPC generated types from the new class library
...
```
Comment (JamesNK) [inline aspnetcore/grpc/troubleshoot.md:660]:
```suggestion
> Many third-party gRPC tools don't support services hosted in subdirectories. Consider finding a way to host gRPC as the root directory.
```

**Review by wadepickett — COMMENTED** [review]
Provided JamesNK's suggestion is added, this looks good to me.  I will wait for @lucasmaj to have a chance to look it over and respond before adding the commits, squash and merge to main.

**Comment by lucasmaj** [comment]
Looks good.

**Pull request merged**